scipy/sparse/linalg/isolve/tests/test_minres.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from numpy.testing import (assert_equal, assert_array_equal, assert_allclose,
+                           assert_raises, run_module_suite, dec)
+from scipy.sparse import diags, eye, lil_matrix
+from scipy.sparse.linalg.isolve import minres
+
+from .test_iterative import assert_normclose
+
+
+def get_sample_problem():
+    # A Random 10 x 10 sparse matrix
+    np.random.seed(1234)
+    matrix = diags(np.random.rand(10), format='csr')
```

Comment:
A diagonal matrix seems too restrictive. Generate a full symmetric matrix (some zeros are possible for sure, considering this is mainly for sparse matrices).